Dockerfile

```diff
@@ -51,7 +51,8 @@ ENV RELAY_FEATURES=${RELAY_FEATURES}
 COPY . .
 
 # Build with the modern compiler toolchain enabled
-RUN scl enable devtoolset-10 llvm-toolset-7.0 -- \
+RUN echo -e "[net]\ngit-fetch-with-cli = true" > $CARGO_HOME/config \
+    && scl enable devtoolset-10 llvm-toolset-7.0 -- \
```

Comment:
This can also be [set on the command line](https://doc.rust-lang.org/cargo/reference/config.html#command-line-overrides), but I'm not sure if we want to change the Makefile for this. Maybe we can just insert a `RUN cargo --config net.git-fetch-with-cli=true fetch` before this command?

Reply:
For reference, in symbolic it's been done like this: https://github.com/getsentry/symbolic/pull/585
Ideally we fix this for the binary and library builds alike.

Reply:
Excuse me, what does this command means (effect) in this case ? `RUN cargo --config net.git-fetch-with-cli=true fetch` @jjbayer

Reply:
@jan-auer  yeah, so that's what this PR does as well, just adds the config for cargo with same option.